kernel/syscall.c

```diff
@@ -3917,9 +4025,9 @@ static long _syscall(void* args_)
             /* the kstack is freed later by next exec or by exit */
             thread->exec_kstack = args->kstack;
 
-            long ret = myst_syscall_execve(filename, argv, envp);
-            /* myst_syscall_execve() only returns on failure */
-            /* when myst_syscall_execve() returns on failure, kstack will be
+            long ret = myst_syscall_execveat(AT_FDCWD, filename, argv, envp, 0);
+            /* myst_syscall_execveat() only returns on failure */
```

Comment:
I would keep both `myst_syscall_execve` and `myst_syscall_execveat`, with the latter calling `myst_get_absolute_path_from_dirfd` and then `myst_syscall_execve`.

Reply:
I think it's more reasonable to only use myst_syscall_execveat, since execve is one of the special case in execveat.

Reply:
I think it's the other way around. This applies to virtually all syscalls with a "at" variant, such as openat, chmodat, etc.

Reply:
Please see my new comment.